### PR TITLE
refactor(2305): make eventscreate theme aware

### LINKED ISSUE #2305: [refactor] Remove direct usage of Theme from "EventsCreate.tsx"
Remove the following

```ts
// TODO: Remove direct usage of Theme
import { preciousPlasticTheme } from 'oa-themes'
const theme = preciousPlasticTheme.styles
```

At the moment we are importing the theme object directly and accessing its style properties. We should avoid doing this and instead make use of `sx` property available on all ThemeUI components to load these values out of the theme definition.

Read more about it here: 
> The `sx` prop lets you add any valid CSS to an element, while using values from your theme to keep styles consistent. You can think of the style object that the `sx` prop accepts as a _superset_ of CSS.
https://theme-ui.com/sx-prop

The aim should be for there to be no visual differences. However there may be circumstances where this would introduce better visual consistency across the UI. Feel free to reach out on [Discord ](https://discord.com/channels/586676777334865928/938781727017558018) if you think you have found one of those scenarios.

Here's an example where the same change has been made, although I would call out that this specific file required more work than most 🤗  https://github.com/ONEARMY/community-platform/pull/2282

### PR DESCRIPTION
**Pull request by bagiyal**
PR Checklist

PR Type

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Developer experience (improves developer workflows for contributing to the project)

## Description

Removes direct usage of the theme object, instead accessing it within the sx property.

## Closes
#2305

### CODE REVIEW

**Review by AlfonsoGhislieri — CHANGES_REQUESTED** [review]
Thanks for the PR @bagiyal, just a few things still missing:
- Please make sure you are using prettier for the formatting, this is required to make sure we have uniformity across the codebase. The linting tests are not pasing ( https://prettier.io/ )
- Please try to give clearer commits/PR titles in future work. 'forget to remove unused thing', is quite vague and not very informative. Something like 'refactor: make eventscreate theme aware' would go a long way.

Thanks for the work and let me know if you have any questions!

**Comment by AlfonsoGhislieri** [comment]
> Sorry man !! Next time I will surely remember these things 🙂

No worries, appreciate the work! Let me know if you need any help with this.

**Comment by bagiyal** [comment]
Yeah i wanna contribute more in precious plastic Give me more chances 😊

File: src/pages/Events/Content/EventsCreate/EventsCreate.tsx [hunk 1]
```diff
@@ -1,3 +1,6 @@
+/** @jsxImportSource theme-ui */
+
```
[the remark this answers is not in the document]
Comment (bagiyal) [reply src/pages/Events/Content/EventsCreate/EventsCreate.tsx:2]: yeah sure

**Review by bagiyal — COMMENTED** [review]
Done changes that you said

**Comment by AlfonsoGhislieri** [comment]
Still have some failing tests here @bagiyal, these are linked to your commit messages. You want to follow convential commit messages, more here: https://www.conventionalcommits.org/en/v1.0.0/

Also conflict needs to be resolved before we can merge this in!

**Comment by AlfonsoGhislieri** [comment]
The tests here are failing still @bagiyal, commit [a4f2765](https://github.com/ONEARMY/community-platform/pull/2470/commits/a4f27651e20446eae52a7d798dbc2d8a4a7480bb) does not adhere to the linting standards. 

This is also getting quite messy with lots of commits. I can suggest two options:
- Use rebase to `squash  `the commits and also rename the old `commit`: https://docs.github.com/en/pull-requests/committing-changes-to-your-project/creating-and-editing-commits/changing-a-commit-message
- Open a new PR if you just find that easier

**Comment by bagiyal** [comment]
@AlfonsoGhislieri i have done changes. Can you check again

File: src/pages/Events/Content/EventsCreate/EventsCreate.tsx [hunk 2]
```diff
@@ -77,11 +75,6 @@ export class EventsCreate extends React.Component<IProps, IState> {
 
     const shouldShowLocationError =
       !isDigitalEvent && isLocationSelected !== undefined && !isLocationSelected
-
-    const _labelStyle = {
-      fontSize: 2,
-      marginBottom: 2,
-    }
```
Comment (AlfonsoGhislieri) [inline src/pages/Events/Content/EventsCreate/EventsCreate.tsx:84]: **Issue(blocking)**: why was this styling removed and similarly from all the sx property of all labels?

**Review by AlfonsoGhislieri — CHANGES_REQUESTED** [review]

**Comment by AlfonsoGhislieri** [comment]
@bagiyal as I said earlier the issue is:
image

There are too many commits for this issue and you still have not fixed the commit message [a4f2765](https://github.com/ONEARMY/community-platform/pull/2470/commits/a4f27651e20446eae52a7d798dbc2d8a4a7480bb).

Please read my message from before on how to solve these issues.

**Comment by AlfonsoGhislieri** [comment]
Closing this PR, I cleaned it up in #2305, you will still have credit for the work @bagiyal

**Pull request closed without merging**